Tasks/Maven/mavencodeanalysis.ts

```diff
@@ -231,36 +238,36 @@ function createSummaryLine(toolName:string, analysisResults:ar.AnalysisResult[])
 }
 
 // Upload build artifacts from all modules
-function uploadBuildArtifactsFromModules(enabledTools:string[], modules:ma.ModuleAnalysis[]) {
+function uploadBuildArtifactsFromModules(enabledTools:string[], modules:ModuleAnalysis[]) {
     enabledTools.forEach((toolName:string) => {
-        modules.forEach((module:ma.ModuleAnalysis) => {
+        modules.forEach((module:ModuleAnalysis) => {
             uploadBuildArtifactsFromModule(toolName, module);
         });
     });
 }
 
 // Copy output files to a staging directory and upload them as build artifacts.
 // Each tool-module combination uploads its own build artifact.
-function uploadBuildArtifactsFromModule(toolName:string, moduleAnalysis:ma.ModuleAnalysis):void {
-    var filesToUpload:string[] = [];
+function uploadBuildArtifactsFromModule(toolName:string, moduleAnalysis:ModuleAnalysis):void {
+    var analysisResult:AnalysisResult;
     if (moduleAnalysis.analysisResults[toolName]) {
-        filesToUpload = moduleAnalysis.analysisResults[toolName].filesToUpload;
+        analysisResult = moduleAnalysis.analysisResults[toolName];
     }
 
-    if (filesToUpload.length < 1) {
+    // If there are no files to upload or there were no violations, return early
+    if (!analysisResult.filesToUpload || analysisResult.filesToUpload.length < 1 || analysisResult.totalViolations < 1) {
```

Comment:
The actual fix - now that we refer to more than one thing from the module's AnalysisResult, we take an instance of that instead.

Then we check that fileToUpload is not null (mostly for robustness), that it contains elements, and that there are violations in the current AnalysisResult. If there are no violations, there isn't any real reason to upload the artifacts.